sklearn/preprocessing/tests/test_data.py

```diff
@@ -1535,9 +1528,17 @@ def _check_one_hot(X, X2, cat, n_features):
     assert_array_equal(toarray(B), toarray(D))
 
 
+def test_one_hot_encoder_string():
+    X = [['cat', 'domestic'], ['wolf', 'wild']]
+    enc = OneHotEncoder()
+    enc.fit(X)
+    Xtr = enc.transform([['cat', 'wild']])
+    assert_array_equal(toarray(Xtr), [[1, 0, 0, 1]])
+
+
 def test_one_hot_encoder_categorical_features():
     X = np.array([[3, 2, 1], [0, 1, 1]])
-    X2 = np.array([[1, 1, 1]])
+    X2 = np.array([[3, 1, 1]])
```

Comment:
I can see it, but I don't get why it's worth fussing over. API change needs to be somewhat conservative to avoid breaking user code as often as possible; and when we break it, to minimise the amount they need to change to achieve the same functionality. I just don't see the current functionality as sufficiently *broken* to justify change.